ibm/service/power/resource_ibm_pi_network_interface.go

```diff
@@ -0,0 +1,356 @@
+				ForceNew:     true,
+				Required:     true,
+				Type:         schema.TypeString,
+				ValidateFunc: validation.NoZeroValues,
+			},
+			Arg_UserTags: {
+				Description: "The user tags attached to this resource.",
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Optional:    true,
+				Set:         schema.HashString,
+				Type:        schema.TypeSet,
+			},
+			// Attributes
+			Attr_CRN: {
+				Computed:    true,
+				Description: "The Network Interface's crn.",
+				Type:        schema.TypeString,
+			},
+			Attr_Instance: {
+				Computed:    true,
+				Optional:    true,
+				Description: "The attached instance to this Network Interface.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						Attr_Href: {
+							Computed:    true,
+							Description: "Link to instance resource.",
+							Type:        schema.TypeString,
+						},
+						Attr_InstanceID: {
+							Computed:    true,
+							Description: "The attached instance ID.",
+							Type:        schema.TypeString,
+						},
+					},
+				},
+				Type: schema.TypeList,
+			},
+			Attr_IPAddress: {
+				Computed:    true,
+				Description: "The ip address of this Network Interface.",
+				Type:        schema.TypeString,
+			},
+			Attr_MacAddress: {
+				Computed:    true,
+				Description: "The mac address of the Network Interface.",
+				Type:        schema.TypeString,
+			},
+			Attr_Name: {
+				Computed:    true,
+				Description: "Name of the Network Interface (not unique or indexable).",
+				Type:        schema.TypeString,
+			},
+			Attr_NetworkInterfaceID: {
+				Computed:    true,
+				Description: "The unique identifier of the network interface.",
+				Type:        schema.TypeString,
+			},
+			Attr_NetworkSecurityGroupID: {
+				Computed:    true,
+				Description: "ID of the Network Security Group the network interface will be added to.",
+				Type:        schema.TypeString,
+			},
+			Attr_Status: {
+				Computed:    true,
+				Description: "The status of the network address group.",
+				Type:        schema.TypeString,
+			},
+		},
+	}
+}
+
+func resourceIBMPINetworkInterfaceCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sess, err := meta.(conns.ClientSession).IBMPISession()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	cloudInstanceID := d.Get(Arg_CloudInstanceID).(string)
+	networkID := d.Get(Arg_NetworkID).(string)
+	networkC := instance.NewIBMPINetworkClient(ctx, sess, cloudInstanceID)
+	var body = &models.NetworkInterfaceCreate{}
+	if v, ok := d.GetOk(Arg_IPAddress); ok {
+		body.IPAddress = v.(string)
+	}
+	if v, ok := d.GetOk(Arg_Name); ok {
+		body.Name = v.(string)
+	}
+	if v, ok := d.GetOk(Arg_UserTags); ok {
+		userTags := flex.FlattenSet(v.(*schema.Set))
+		body.UserTags = userTags
+	}
+	networkInterface, err := networkC.CreateNetworkInterface(networkID, body)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	networkInterfaceID := *networkInterface.ID
```

Comment:
better add a nil pointer check

Reply:
The ID is a required field from SB, if one is missing that means an error has occurred during creation and an err will be return. So we will go pass that point therefore a nil pointer check is not necessary here.

Reply:
I have seen nil references and no err message in DHCP work during its incubation period. But I see nil check on response payload in the power go client. We should be good here.